src/components/TodoList/TodoList.tsx

```diff
@@ -1,51 +1,95 @@
-      </tr>
-    </thead>
+type Props = {
+  todos: Todo[];
+  visible: boolean;
+  setTodoOnClick: (arg: Todo) => void;
+  showModal: (arg: boolean) => void;
+};
 
-    <tbody>
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-      <tr data-cy="todo" className="has-background-info-light">
-        <td className="is-vcentered">2</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+export const TodoList: React.FC<Props> = ({
+  todos,
+  visible,
+  setTodoOnClick,
+  showModal,
+}) => {
+  const handleView = (id: number) => {
+    const newTodo = todos.find(todo => todo.id === id);
+
+    if (newTodo) {
+      setTodoOnClick(newTodo);
+      showModal(true);
+    }
+  };
+
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>
             <span className="icon">
-              <i className="far fa-eye-slash" />
+              <i className="fas fa-check" />
             </span>
-          </button>
-        </td>
-      </tr>
+          </th>
+          <th>Title</th>
+          <th> </th>
+        </tr>
+      </thead>
 
-      <tr data-cy="todo" className="">
+      <tbody>
+        {todos.map(todo => {
+          const { id, title, completed } = todo;
+
+          return (
+            <tr data-cy="todo" className="" key={id}>
+              <td className="is-vcentered">{id}</td>
+              <td className="is-vcentered">
+                {completed && (
+                  <span className="icon" data-cy="iconCompleted">
+                    <i className="fas fa-check"></i>
+                  </span>
+                )}
+              </td>
+              <td className="is-vcentered is-expanded">
+                <p
+                  className={classNames({
+                    'has-text-danger': !completed,
+                    'has-text-success': completed,
+                  })}
+                >
+                  {title}
+                </p>
+              </td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  data-cy="selectButton"
+                  className="button"
+                  type="button"
+                  onClick={() => handleView(id)}
```

Comment:
The `onClick` handler can be simplified by directly passing `handleView` with the `id` parameter: `onClick={() => handleView(id)}` is already optimal, but ensure `handleView` is correctly handling the logic.